### ISSUE TITLE
[Question] Custom date on catch-up event

### ISSUE THREAD

**Issue by MaximeGrbd** [jhuckaby/Cronicle#532]
Hi, 

I have some difficulties with the catch-up feature.

Here is my problem :
I have a job that is launched every day, the job execute a shell script via the shell plugin. This shell script execute some stuff with a date as a parameter (date that is create like this : NOW=$(date +"%Y-%m-%d")).
Then if I enable the catch-up mode and set the event clock to 1 week before, it will launch 7 jobs like I want but with the 7 same date, the date is the same to when I enable the catch-up mode.

I would like the date to change and match the missed execution for each job.

For example :
for the date 2022-09-28, I enable the catch-up mode with the event clock set to  2022-09-22,
it will execute 7 jobs with the date 2022-09-28 as the parameter. 
I would like to have these job with the date paramater as : 
1st job : date = 2022-09-22 
2nd job :  date = 2022-09-23
 . . . 
Last job : date = 2022-09-28

I don't know if it's possible or if I missed something.

Thanks for your help.

**Comment by jhuckaby** [comment]
Yup, this is handled via JSON input (and also an environment variable).  See the docs here:

https://github.com/jhuckaby/Cronicle/blob/master/docs/Plugins.md#json-input

> The `now` property is special.  It may or may not be the "current" time, depending on how and why the job started.  Meaning, if the associated event is set to [Run All Mode](https://github.com/jhuckaby/Cronicle/blob/master/docs/WebUI.md#run-all-mode) it is possible that a previous job couldn't run for some reason, and Cronicle is now "catching up" by running missed jobs.  In this situation, the `now` property will be set to the time when the event *should have* run.  For example, if you have a Plugin that generates daily reports, you'll need to know *which day* to run the report for.  Normally this will be the current date, but if a day was missed for some reason, and Cronicle is catching up by running yesterday's job, you should use the `now` time to determine which day's data to pull for your report.

This is also available via `JOB_NOW` environment variable (see [Job Environment Variables](https://github.com/jhuckaby/Cronicle/blob/master/docs/Plugins.md#job-environment-variables)), which is better for shell scripts, and is in Epoch seconds.  You can use this to construct your date/time strings.

```sh
NOW=$(date -r $JOB_NOW +"%Y-%m-%d")
```

**Comment by MaximeGrbd** [comment]
Thanks for these informations ! 
I will look into it !

**Comment by MaximeGrbd** [comment]
In the same way, is it possible to transmit this date to other job ?

I have mutliple job that execute in cascade. Only the first job is schedule with calendar date, all other job are set as chain reaction (job 1 => job2 if job1 is success => job3 if job2 is success).
But even if I set the date as ```NOW=$(date -d @$JOB_NOW +"%Y-%m-%d")``` in all job, only the first job that is scheduled has a past date. All job that are execute by reaction chain has current date. 

I can avoid this by manually schedule all job and then execute them manually one wave after another, but if it's could completely be automatic it would be great. 

Maybe it's possible with handling JSON input or some  specific configuration ?

Thanks for your help.

**Comment by jhuckaby** [comment]
Hmmm, that's a little tricky, but it can be done, using the [Chain Data](https://github.com/jhuckaby/Cronicle/blob/master/docs/Plugins.md#chain-data) system.

I haven't tried this with shell scripts, but you should be able to push the custom date into the chain data, and then pull it back out on subsequent jobs.

If I can find some time, I'll try to whip up an example.

**Comment by MaximeGrbd** [comment]
Thanks for these advices, I will try to figure out using chain data. 
If you can indeed whip an example it would be great !

**Comment by MaximeGrbd** [comment]
I succeed doing what I wanted with the chain Data and JSON output, it's work well !

Thanks for your work and your help!
